f8a_worker/workers/githuber.py

```diff
@@ -28,41 +37,41 @@ def create_test_instance(cls, repo_name, repo_url):
         instance._repo_url = repo_url
         return instance
 
-    @staticmethod
-    def _retry_no_cached(call, sleep_time=2, retry_count=10):
-        """ Deal with cached results from GitHub as PyGitHub does not check this
-
-        https://developer.github.com/v3/repos/statistics/#a-word-about-caching
-        """
-        result = None
-
-        for _ in range(retry_count):
-            result = call()
-            if result:
-                break
-            time.sleep(sleep_time)
-
-        return result
+    def _get_response(self, url, sleep_time=2, retry_count=10):
+        try:
+            # https://developer.github.com/v3/repos/statistics/#a-word-about-caching
+            for _ in range(retry_count):
+                response = requests.get(url, headers=self._headers)
+                response.raise_for_status()
+                response = response.json()
+                if response:
+                    return response
+                time.sleep(sleep_time)
+        except requests.exceptions.HTTPError as err:
+            self.log.error(
+                "Failed to get results from {url} with {err}".format(url=url, err=err)
+            )
+            return {}
 
     @classmethod
     def _get_last_years_commits(cls, repo):
-        activity = cls._retry_no_cached(repo.get_stats_commit_activity)
+        activity = self._get_response(urljoin(repo_url + '/', "stats/commit_activity"))
```

Comment:
This will fail as this is class method and there is no `repo_url` defined in this scope.